### ISSUE TITLE
Upvotes are bugged on Live

### ISSUE THREAD

**Issue by MichaelJGarlotteII** [MichaelJGarlotteII/testProject#17]
Today (3/27/2020) when we pushed the new upvote feature to production, I found that the feature was working great until I tried to use it on Chrome and Safari. After attempting to upvote on those browsers, the feature failed to work altogether regardless of browser.

**Comment by grahamjamesr7** [comment]
Michael -- can you post the related logs here so I can reference them? Can you also let me know the circumstances surrounding your test? ie: were you logged in? Were you performing the first upvote? Did browsers show upvotes successfully on page load but fail to allow upvotes and update the count? 

I need a bit more detail before I can start work. Thanks!

**Comment by MichaelJGarlotteII** [comment]
When conducting the test I was logged in as a brand new user. When conducting this test on Firefox, the upvote was correctly distributed and shown as follows:

<img width="960" alt="Capture1" src="https://user-images.githubusercontent.com/62119133/77797015-52f08880-703e-11ea-8945-e6b2aafc215f.PNG">

When upvoting a post that already has upvotes, the post has 3 upvotes and seemingly refuses to apply any more. Refreshing the page does not rectify the upvote count. This attempt does not produce an error, as shown below:

<img width="960" alt="Capture2" src="https://user-images.githubusercontent.com/62119133/77797212-b1b60200-703e-11ea-89a6-3ebad6352596.PNG">

When attempting the same thing on Chrome, the following occurs for each case:
<img width="960" alt="Capture3" src="https://user-images.githubusercontent.com/62119133/77797508-2e48e080-703f-11ea-8f18-1270a435f881.PNG">
<img width="960" alt="Capture4" src="https://user-images.githubusercontent.com/62119133/77797514-31dc6780-703f-11ea-9626-07ee17ef7a84.PNG">
The errors are

> Access to XMLHttpRequest at 'http://michaeljgarlotteii.com:5000/login?next=%2Fapi%2Fpost%2F2%2Fupvote' (redirected from 'http://michaeljgarlotteii.com:5000/api/post/2/upvote') from origin 'http://www.michaeljgarlotteii.com:5000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
jquery-3.3.1.min.js:2 
>GET http://michaeljgarlotteii.com:5000/login?next=%2Fapi%2Fpost%2F2%2Fupvote net::ERR_FAILED

Refreshing does nothing to rectify the issue.
I imagine the same issue is happening when I attempt from safari but afaik mobile Safari doesn't have a dev console I can work with.

**Comment by MichaelJGarlotteII** [comment]
Following up on the offline discussions, here are the relevant messages from the console. This message occurs exclusively on chrome; however, the upvote functionality was also bugged in safari on mobile (was not able to access a console on that browser).

![Capture7](https://user-images.githubusercontent.com/62119133/78941974-547a7180-7a7e-11ea-95c8-accd4623f9a7.PNG)

**Comment by grahamjamesr7** [comment]
As discussed in #20, the issue turned out to be that my solution of using the `@cross_origin()` annotation wasn't sufficient on its own to enable CORS for that endpoint. After a brief perusal of stack overflow by myself and @MichaelJGarlotteII, we discovered that all that was missing was an initialization of the CORS module. After we initialized CORS and allowed cross origin requests across all `/api/` endpoints, the issue was resolved.
